.github/workflows/reusable-e2e.yaml

```diff
@@ -0,0 +1,191 @@
+        uses: actions/download-artifact@v3
+        with:
+          name: test-operator
+          path: /tmp
+      - name: start minikube
+        uses: medyagh/setup-minikube@master
+        with:
+          container-runtime: containerd
+      - run: minikube image load /tmp/test-operator.tar
+      - run: minikube image list
+      - name: setup operator
+        run: |
+          target/debug/mirrord operator setup --accept-tos --license-path ./tests/test-license.pem | kubectl apply -f -              
+          kubectl get pods
+      - name: wait for operator status
+        run: |
+          end_time=$((SECONDS+60))
+          log_file="error.log"
+          rm -f "$log_file"
+
+          while [ $SECONDS -lt $end_time ]; do
+            if target/debug/mirrord operator status 2>> "$log_file"; then              
+              break
+            fi
+            kubectl describe pods -n mirrord
+            sleep 1
+          done
+
+          if [ $SECONDS -ge $end_time ]; then
+            echo "Timed out!, error messages:"            
+            cat "$log_file"            
+          fi
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+      - run: npm install express
+      - uses: actions/setup-python@v3
+      - run: pip3 install flask
+      - run: pip3 install fastapi
+      - run: pip3 install uvicorn[standard]
+      - run: sudo apt-get update && sudo apt-get install -y curl
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.18.0"
+      - run: |
+          go version
+      - run: | # Build Go test apps.
+          ./scripts/build_go_apps.sh 18
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.19.0"
+      - run: |
+          go version
+      - run: | # Build Go test apps.
+          ./scripts/build_go_apps.sh 19
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.20.0-rc.3"
+      - run: |
+          go version
+      - run: | # Build Go test apps.
+          ./scripts/build_go_apps.sh 20
+      - run: |
+          cd tests/rust-e2e-fileops
+          cargo build
+      - run: |
+          cd tests/rust-unix-socket-client
+          cargo build
+      - run: |
+          cd tests/rust-bypassed-unix-socket
+          cargo build
+      - name: download agent image
+        uses: actions/download-artifact@v3
+        with:
+          name: test
+          path: /tmp
+      - run: minikube image load /tmp/test.tar
+      # By running the test of the targetless agent first, we prove it works on an empty cluster without any pods.
+      - name: Run targetless E2E test.
+        run: cargo test -p tests targetless
+      - name: Run all E2E test
+        run: cargo test -p tests -- --test-threads=6
+      - name: Collect logs
+        if: ${{ failure() }}
+        run: |
+          kubectl describe pods
+          docker exec minikube find /var/log/pods -print -exec cat {} \;
+
+  mirrord_e2e:    
```

Comment:
Maybe it's better to split this entire step into 2 steps one for mirrord build and another for e2e and leave the creation of the cluster for the "reuser" (on second step assume you have a cluster running already)